Dynamitey/Internal/Optimization/InvokeHelper-Regular.cs

```diff
@@ -446,36 +446,17 @@ internal static object InvokeGetCallSite(object target, string name, Type contex
                 if (staticContext) //CSharp Binder won't call Static properties, grrr.
                 {
                     var tStaticFlag = CSharpBinderFlags.None;
-                    if ((target is Type && ((Type)target).GetTypeInfo().IsPublic))
-                    {
-                        tBinder = () => Binder.InvokeMember(tStaticFlag, "get_" + name,
-                                                            null,
-                                                            context,
-                                                            new List<CSharpArgumentInfo>
-                                                                {
-                                                                    CSharpArgumentInfo.Create(
-                                                                        CSharpArgumentInfoFlags.IsStaticType |
-                                                                        CSharpArgumentInfoFlags.UseCompileTimeType,
-                                                                        null)
-                                                                });
-
-                        tBinderType = typeof (InvokeMemberBinder);
-                        tKnownType = KnownMember;
-                    }
```

Comment:
This binder only works for public type. The `else` only works for private type when a property is set-first...